### ISSUE TITLE
Deprecate the PHAR metadata

### ISSUE THREAD

**Issue by theofidry** [box-project/box#1152]
## Feature Request

Setting the PHAR metadata was supported because it was a PHAR option. I am however not convinced this is a good thing as:

- I never saw a good practical example.
- It used to expose to security risks and in a way still does. And by design there is no control about how the metadata is set.

One notable exception is @llaville's https://github.com/llaville/box-manifest. That being said the core of the project is to provide a manifest in different formats and I am not convinced using the metadata here is a good idea (and Box should offer another way than relying on the metadata setting to achieve this).

Thoughts @sebastianbergmann @theseer? Do you have a use case for the PHAR metadata?

**Comment by theseer** [comment]
Back when I introduced phar building support to [Autoload](https://github.com/theseer/Autoload), I looked into it and already found it rather useless. 

When we started on [Phive](https://github.com/phar-io/phive), I revisited that topic as we wanted a "reliable" source on what would be within the phar archive and what would be its system requirements. Since we did not find anything we could use, we invented our on [Manifest](https://github.com/phar-io/manifest). It's stored as a "regular" file within the phar and does not use the metadata API.

**TL;DR: No**

Given that with SBOMs on the rise as a defacto industry standard, we might have a look into having out-of-the-box (no pun intended) support as well as a standard way to query it. We might also ask who ever is currently maintaining `ext/phar` to add support for SBOMs.
